mongo/options/findoptions.go

```diff
@@ -563,6 +563,12 @@ type FindOneAndReplaceOptions struct {
 	// If true, a new document will be inserted if the filter does not match any documents in the collection. The
 	// default value is false.
 	Upsert *bool
+
+	// The index to use for the operation. This should either be the index name as a string or the index specification
+	// as a document. The default value is nil, which means that no hint will be sent. This option is only valid for
+	// MongoDB versions >= 4.4. MongoDB versions >= 4.2 will report an error if this option is set. For MongoDB versions
```

Comment:
This reads funny... First it says that MongoDB >= 4.4 supports hint.  Then it says MongoB >= 4.2 will error.  That's not internally consistent.  Unfortunately that's what's in the spec though, so I imagine everyone is using this text now.   But I think it would be clearer to say:

     MongoDB 4.2 will report an error  ...

Reply:
Done.